cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/service/InMemoryServiceProgramRunner.java

```diff
@@ -123,11 +122,13 @@ public ProgramController run(Program program, ProgramOptions options) {
 
   @Override
   public ProgramOptions createComponentOptions(String name, int instanceId, int instances, RunId runId,
-                                               Arguments userArguments) {
-    Map<String, String> options = ImmutableMap.of(ProgramOptionConstants.INSTANCE_ID, Integer.toString(instanceId),
-                                                  ProgramOptionConstants.INSTANCES, Integer.toString(instances),
-                                                  ProgramOptionConstants.RUN_ID, runId.getId(),
-                                                  ProgramOptionConstants.HOST, host);
-    return new SimpleProgramOptions(name, new BasicArguments(options), userArguments);
+                                               ProgramOptions options) {
+    Map<String, String> systemOptions = Maps.newHashMap();
```

Comment:
I don't think this should change anything, but to keep the behavior the same as before, can we use ``ImmutableMap.Builder`` here instead of a ``newHashmap()``?